plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/AppEngineWebXmlValidator.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.validation;
+
+import com.google.cloud.tools.eclipse.appengine.facets.WebProjectUtil;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.io.ByteStreams;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.Map;
+import java.util.Stack;
+import javax.xml.parsers.ParserConfigurationException;
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.Path;
+import org.eclipse.wst.validation.AbstractValidator;
+import org.eclipse.wst.validation.ValidationEvent;
+import org.eclipse.wst.validation.ValidatorMessage;
+import org.eclipse.wst.validation.ValidationResult;
+import org.eclipse.wst.validation.ValidationState;
+import org.xml.sax.SAXException;
+import org.xml.sax.SAXParseException;
+
+/**
+ * Validator for appengine-web.xml
+ */
+public class AppEngineWebXmlValidator extends AbstractValidator {
+  
+  private static final Logger logger = Logger.getLogger(
+      AppEngineWebXmlValidator.class.getName());
+  private IProject project;
+  
+  @Override
+  public void validationStarting(IProject project,
+      ValidationState state, IProgressMonitor monitor) {        
+    this.project = project;
+  }
+  
+  /**
+   * Locates appengine-web.xml in project.
+   */
+  @Override
+  public ValidationResult validate(ValidationEvent event, ValidationState state,
+      IProgressMonitor monitor) {
+    IResource resource = event.getResource();
+    IFile file = WebProjectUtil.findInWebInf(project, new Path("appengine-web.xml"));
```

Comment:
> It shouldn't return null, because the file exists or validation wouldn't have started.

Makes sense. Speaking of which, I wonder if `event.getResource()` could be already `appengine-web.xml`? (Have no idea; just a thought.)